lib/reamaze_api/response_middleware.rb

```diff
@@ -0,0 +1,65 @@
+module ReamazeAPI
+  class ResponseMiddleware < Faraday::Response::Middleware
+    # Public: Logs the API request when a Faraday request starts.
+    #
+    # env - Faraday::Env instance
+    #
+    # Returns nothing.
+    def call(env)
+      log("Request", env_info(env))
+      super
+    end
+
+    # Public: Logs the API response when a Faraday request completes.
+    #
+    # env - Faraday::Env instance
+    #
+    # Returns nothing.
+    def on_complete(env)
+      log("Response", env_info(env))
+
+      if error = ReamazeAPI::Error.from_response(env)
+        raise error
+      end
+    end
+
+    private
+
+    # Private: Logs the given action/message to the HelixCore.logger.
+    #
+    # action  - "Request" or "Response"
+    # message - Log message
+    #
+    # Returns nothing.
+    def log(action, message)
+      return unless logger = ReamazeAPI.config.logger
+
+      message = "#{action} | #{message}"
+
+      if logger.respond_to?(:call)
+        logger.call message
+      else
+        logger.info message
+      end
+    end
+
+    # Private: Strips sensitive info from the response body.
+    #
+    # env - Faraday::Env instance
+    #
+    # Returns a String.
+    def env_info(env)
+      words       = %w(body).join('|')
```

Comment:
Are there any other fields besides body you can think of that people would want to filter?